src/order/filter/basic.lean

```diff
@@ -2010,11 +2014,106 @@ lemma tendsto_at_top_mono [preorder β] (l : filter α) :
   monotone (λ f : α → β, tendsto f l at_top) :=
 λ f₁ f₂ h, tendsto_at_top_mono' l $ univ_mem_sets' h
 
+/-!
+## Sequences
+-/
+
 @[nolint ge_or_gt] -- see Note [nolint_ge]
 lemma map_at_top_inf_ne_bot_iff [semilattice_sup α] [nonempty α] {F : filter β} {u : α → β} :
   (map u at_top) ⊓ F ≠ ⊥ ↔ ∀ U ∈ F, ∀ N, ∃ n ≥ N, u n ∈ U :=
 by simp_rw [inf_ne_bot_iff_frequently_right, frequently_map, frequently_at_top] ; trivial
 
+lemma extraction_of_frequently_at_top' {P : ℕ → Prop} (h : ∀ N, ∃ n > N, P n) :
+  ∃ φ : ℕ → ℕ, strict_mono φ ∧ ∀ n, P (φ n) :=
+begin
+  choose u hu using h,
+  cases forall_and_distrib.mp hu with hu hu',
+  exact ⟨u ∘ (nat.rec 0 (λ n v, u v)), strict_mono.nat (λ n, hu _), λ n, hu' _⟩,
+end
+
+lemma extraction_of_frequently_at_top {P : ℕ → Prop} (h : ∃ᶠ n in at_top, P n) :
+  ∃ φ : ℕ → ℕ, strict_mono φ ∧ ∀ n, P (φ n) :=
+begin
+  rw frequently_at_top' at h,
+  exact extraction_of_frequently_at_top' h,
+end
+
+lemma extraction_of_eventually_at_top {P : ℕ → Prop} (h : ∀ᶠ n in at_top, P n) :
+  ∃ φ : ℕ → ℕ, strict_mono φ ∧ ∀ n, P (φ n) :=
+extraction_of_frequently_at_top (eventually.frequently at_top_ne_bot h)
+
+@[nolint ge_or_gt] -- see Note [nolint_ge]
+lemma exists_le_of_tendsto_at_top [semilattice_sup α] [preorder β] {u : α → β}
+  (h : tendsto u at_top at_top) : ∀ a b, ∃ a' ≥ a, b ≤ u a' :=
+begin
+  intros a b,
+  have : ∀ᶠ x in at_top, a ≤ x ∧ b ≤ u x := inter_mem_sets (mem_at_top a) (h $ mem_at_top b),
+  haveI : nonempty α := ⟨a⟩,
+  rcases this.exists at_top_ne_bot with ⟨a', ha, hb⟩,
+  exact ⟨a', ha, hb⟩
+end
+
+@[nolint ge_or_gt] -- see Note [nolint_ge]
+lemma exists_lt_of_tendsto_at_top [semilattice_sup α] [preorder β] [no_top_order β]
+  {u : α → β} (h : tendsto u at_top at_top) : ∀ a b, ∃ a' ≥ a, b < u a' :=
+begin
+  intros a b,
+  cases no_top b with b' hb',
+  rcases exists_le_of_tendsto_at_top h a b' with ⟨a', ha', ha''⟩,
+  exact ⟨a', ha', lt_of_lt_of_le hb' ha''⟩
+end
+
+@[nolint ge_or_gt] -- see Note [nolint_ge]
+lemma high_scores [linear_order β] [no_top_order β] {u : ℕ → β}
```

Comment:
Please add (almost identical) docstrings to this and the next lemmas.

Reply:
Or a module section docstring to this section.

Reply:
I'm sorry, I don't understand what you want here. The statement looks really easy to read. What else do you want  to add?

Reply:
I agree it's pretty straightforward, but how about
```
/--
If `u` is a sequence which is unbounded above,
then after any point, it reaches a value strictly greater than all previous values.
-/
```
and
```
/--
If `u` is a sequence which is unbounded above,
then it `frequently` reaches a value strictly greater than all previous values.
-/
```
for the next lemma?

(While I don't think these ones particularly need paraphrases, but plenty often I meet statements in mathlib that I wish had an informal statement to help decode, so I guess it doesn't hurt to set good examples. :-)

Reply:
Ok, I used Scott's docstrings.

Reply:
I wouldn't have guessed this name... but I don't really have a suggestion for a better name...
`exists_forall_le_of_tendsto_at_top` is the best I can come up with, and I don't think it's good.